### PR TITLE
Serialization strategies

### PR DESCRIPTION
**Pull request by mccorby**
This PR enables different strategies to serialise/deserialise tensors.
Right now the serialisation of the tensors is delegated to Torch but the serialised objects are not compatible with other platforms

This PR includes a strategy to seder the tensors using npy format

### CODE REVIEW

File: syft/serde.py [hunk 1]
```diff
@@ -157,6 +155,74 @@ def deserialize(
     return _detail(worker, simple_objects)
 
 
+def serialize_tensor(tensor) -> bin:
```
Comment (robert-wagner) [inline syft/serde.py]: This is clever

File: syft/serde.py [hunk 2]
```diff
@@ -27,35 +27,33 @@
 
 By default, we serialize using msgpack and compress using lz4.
 """
-
+import io
+import zstd
```
Comment (robert-wagner) [inline syft/serde.py]: Could you move this import down to below torch

File: syft/serde.py [hunk 3]
```diff
@@ -157,6 +155,74 @@ def deserialize(
...
+    """
+    return torch_tensor_deserializer(tensor_bin)
+
+
+def numpy_tensor_serializer(tensor: torch.Tensor) -> bin:
+    """Strategy to serialize a tensor using numpy npy format.
+    If tensor requires to calculate gradients, it will detached.
+    """
+    if tensor.requires_grad:
+        print(
```
Comment (robert-wagner) [inline syft/serde.py]: Rather than printing this, I think we should raise it as a warning ie warnings.warn

File: examples/tutorials/Part 2 - Intro to Federated Learning.ipynb [hunk 4]
```diff
@@ -235,7 +234,7 @@
     "            loss.backward()\n",
     "\n",
     "            # 5) change those weights\n",
-    "            opt.step(data.shape[0])\n",
+    "            opt.step()\n",
```
Comment (robert-wagner) [inline examples/tutorials/Part 2 - Intro to Federated Learning.ipynb]: Why was this changed?

File: syft/serde.py (hunk 1 above)
Comment (robert-wagner) [reply syft/serde.py]: I think we might want to make this and deserialize tensor private functions as the user should not be calling them

File: examples/tutorials/Part 2 - Intro to Federated Learning.ipynb (hunk 4 above)
Comment (mccorby) [reply examples/tutorials/Part 2 - Intro to Federated Learning.ipynb]: Leftover from the fork. Removed

**Comment by mccorby** [comment]
@robert-wagner All checks passed. Addressed your comments
